src/dataServices/dynamoDbHelper.ts

```diff
@@ -38,19 +57,12 @@ export default class DynamoDbHelper {
     }
 
     async getMostRecentValidResource(resourceType: string, id: string): Promise<GenericResponse> {
-        const params = DynamoDbParamBuilder.buildGetResourcesQueryParam(id, 2);
-        let item = null;
-        const result = await this.dynamoDb.query(params).promise();
-        const items = result.Items
-            ? result.Items.map((ddbJsonItem: any) => DynamoDBConverter.unmarshall(ddbJsonItem))
-            : [];
-        if (items.length === 0) {
-            throw new ResourceNotFoundError(resourceType, id);
-        }
+        const items = await this.getMostRecentResources(resourceType, id);
         const latestItemDocStatus = items[0][DOCUMENT_STATUS_FIELD];
         if (latestItemDocStatus === DOCUMENT_STATUS.DELETED) {
             throw new ResourceNotFoundError(resourceType, id);
         }
+        let item: any = {};
         // If the latest version of the resource is in PENDING, grab the previous version
         if (latestItemDocStatus === DOCUMENT_STATUS.PENDING && items.length > 1) {
             // eslint-disable-next-line prefer-destructuring

```

Comment:
Is this `if` block valid? What if the document status === PENDING AND items.length === 1? we would return a PENDING resource as valid

Reply:
Good catch. I updated the logic to handle that case. The logic now explicitly looks for resources in valid status (Available, Pending_Delete, Locked). If the latest resource is in Pending, and there are more than one resource, then return the second latest resource. 

If none of these case apply, return a `ResourceNotFound` exception.